src/core/server/data/context.ts

```diff
@@ -0,0 +1,28 @@
+import { Db } from "mongodb";
+
+import { AugmentedRedis } from "coral-server/services/redis";
+
+export interface MongoContext {
+  readonly main: Db;
+  readonly archive: Db;
+}
+
+interface DataContextOptions {
+  mongoLive: Db;
+  mongoArchive: Db;
```

Comment:
You could possibly embed this a bit further to simplify the access:

```suggestion
  mongo: MongoContext;
```

Reply:
Alright, this and your other comment makes me feel like I will go all in and refactor this context through the whole code base. No worries, I hear you and it will be done.